pkg/listening-sockets/listening-sockets.go

```diff
@@ -0,0 +1,309 @@
+package listeningsockets
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/openshift-kni/commatrix/pkg/client"
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/types"
+	"github.com/openshift-kni/commatrix/pkg/utils"
+)
+
+const (
+	localAddrPortFieldIdx = 3
+	interval              = time.Millisecond * 500
+	duration              = time.Second * 5
+)
+
+type ConnectionCheck struct {
+	*client.ClientSet
+	podUtils            utils.UtilsInterface
+	customEntriesPath   string
+	customEntriesFormat string
+	format              string
+	destDir             string
+	env                 types.Env
+	deployment          types.Deployment
+	nodesRolesMap       map[string][]string
+}
+
+func NewCheck(c *client.ClientSet, podUtils utils.UtilsInterface, customEntriesPath, customEntriesFormat, format, destDir string, env types.Env, deployment types.Deployment) (*ConnectionCheck, error) {
+	nodeList := &corev1.NodeList{}
+	err := c.List(context.TODO(), nodeList)
+	if err != nil {
+		return nil, err
+	}
+
+	nodesRolesMap := map[string][]string{}
+	for _, node := range nodeList.Items {
+		nodesRolesMap[node.Name] = types.GetNodeRoles(&node)
+	}
+
+	return &ConnectionCheck{
+		c,
+		podUtils,
+		customEntriesPath,
+		customEntriesFormat,
+		format,
+		destDir,
+		env,
+		deployment,
+		nodesRolesMap,
+	}, nil
+}
+
+func (cc *ConnectionCheck) GenerateSS() (*types.ComMatrix, []byte, []byte, error) {
+	nodesComDetails := []types.ComDetails{}
+	err := cc.podUtils.CreateNamespace(consts.DefaultDebugNamespace)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	defer func() {
+		err := cc.podUtils.DeleteNamespace(consts.DefaultDebugNamespace)
+		if err != nil {
+			fmt.Printf("failed to delete debug namespace: %v", err)
+		}
+	}()
+	var ssOutTCP, ssOutUDP []byte
+	nLock := &sync.Mutex{}
+	g := new(errgroup.Group)
+	for nodeName := range cc.nodesRolesMap {
+		name := nodeName
```

Comment:
why u are doing that and not stay with the nodeName?

Reply:
you can't because you use that in a go routine